### PR TITLE
clen up buddy check mess

### PR DESCRIPTION
**Pull request by grosser**
- test all code
 - fail on invalid numbers
 - make grace_period and time_limit both return seconds
 - deprecate BYPASS_JIRA_EMAIL in favor of comma separated BYPASS_EMAIL
 - remove tests that are already covered by other classes

### CODE REVIEW

File: lib/buddy_check.rb [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
-  def grace_period
-    (ENV["BUDDY_CHECK_GRACE_PERIOD"].presence || "4").to_i.hours
-  end
-
-  def bypass_jira_email_address
-    ENV["BYPASS_JIRA_EMAIL"]
-  end
+    # how long can the same commit be deployed ?
+    def grace_period
+      Integer(ENV["BUDDY_CHECK_GRACE_PERIOD"].presence || "4").hours
```
Comment (sandlerr) [inline lib/buddy_check.rb]: why Integer over to_i?

File: test/lib/buddy_check_test.rb [hunk 2]
```diff
@@ -1,130 +1,90 @@
...
+    it "is enabled when 1" do
+      with_env BUDDY_CHECK_FEATURE: "1" do
+        assert BuddyCheck.enabled?
+      end
+    end
 
-    stage.stubs(:production?).returns(true)
-    service.expects(:confirm_deploy!).once
+    it "is disabled when not 1" do
+      with_env BUDDY_CHECK_FEATURE: "true" do
```
Comment (sandlerr) [inline test/lib/buddy_check_test.rb]: might confuse people, maybe crash instead?

**Comment by sandlerr** [comment]
somehow deleting a lot of tests and increasing coverage, awesome :D

**Comment by grosser** [comment]
Blows up when the value is not a number ... See tests

**Comment by grosser** [comment]
Hmm true ... Future pr ... For now keeping old behavior

**Comment by grosser** [comment]
Deleting duplicate tests ;)

**Pull request merged**